src/main.js

```diff
@@ -97,34 +81,96 @@ function clearSuggestions() {
 }
 
 function showForm() {
+  var form = document.querySelector(".footer");
   form.classList.toggle("hidden");
+  hideRecipeButton();
 }
 
 function displayNewUserIdea() {
-  var userIdea = recipeTypeInput.value.toLowerCase();
-  if (userIdea === 'side') {
+  var userType = recipeTypeInput.value.toLowerCase();
+  if (userType === 'side') {
     newUserSide();
-  } else if (userIdea === 'main dish') {
+  } else if (userType === 'main') {
     newUserMainDish();
-  } else if (userIdea === 'dessert') {
+  } else if (userType === 'dessert') {
     newUserDessert();
   }
+  resetNameForm();
 }
 
 function newUserSide() {
-  sides.push(recipeNameInput.value);
-  suggested = sides[sides.length - 1];
+  var newSide = recipeNameInput.value
+  sides.push(newSide);
+  userSides.push(newSide);
+  suggested = userSides[userSides.length - 1];
+  takeToAllRecipes();
   displaySuggestion();
 }
 
 function newUserMainDish() {
-  mains.push(recipeNameInput.value);
-  suggested = mains[mains.length - 1];
+  var newMain = recipeNameInput.value
+  mains.push(newMain);
+  userMains.push(newMain);
+  suggested = userMains[userMains.length - 1];
+  takeToAllRecipes();
   displaySuggestion();
 }
 
 function newUserDessert() {
-  desserts.push(recipeNameInput.value);
-  suggested = desserts[desserts.length - 1];
+  var newDesert = recipeNameInput.value
+  desserts.push(newDesert);
+  userDesserts.push(newDesert);
+  suggested = userDesserts[userDesserts.length - 1];
+  takeToAllRecipes();
   displaySuggestion();
 }
+
+function displayUserWholeMeal() {
+  uncheckRadioButton();
+  getUserRecipeName();
+  if (suggested.includes(undefined)) {
+    suggested = `Please enter a Side, Main Dish and Dessert in the form below`;
+    displaySuggestion();
+  } else {
+    displaySuggestion();
+  }
+  hideRecipeButton();
+}
+
+function getUserRecipeName() {
+  var userMeal = new Meal(
+    userSides[getRandomFoods(userSides)],
+    userMains[getRandomFoods(userMains)],
+    userDesserts[getRandomFoods(userDesserts)],
+  )
+  suggested = `${userMeal.main} with a side of\n${userMeal.side} and\n${userMeal.dessert}`;
+}
+
+function resetNameForm() {
```

Comment:
I only use this function one time. Is it better like this, or to just put it into the function that calls it instead of writing it here and then calling it?